.cloudbees/workflows/workflow.yaml

```diff
@@ -11,7 +11,66 @@ permissions:
   id-token: write
 
 jobs:
+  build-base-kaniko-image:
+    steps:
+    - name: Checkout this repo (for workflow, scripts, etc.)
+      uses: cloudbees-io/checkout@v1
+
+    - name: Checkout Kaniko upstream repo into subdir
+      uses: cloudbees-io/checkout@v1
+      with:
+        repository: chainguard-dev/kaniko
+        path: kaniko-src
+
+    - name: Login to AWS
+      uses: cloudbees-io/configure-aws-credentials@v1
+      with:
+        aws-region: us-east-1
+        role-to-assume: ${{ vars.oidc_staging_iam_role }}
+        role-duration-seconds: "3600"
+
+    - name: Configure container registry for ECR
+      uses: cloudbees-io/configure-ecr-credentials@v1
+
+    - name: Append GOFLAGS and generate diff
+      uses: docker://golang:1.24.0
+      run: |
+        cd kaniko-src
+        git config user.name "ci"
+        git config user.email "ci@example.com"
+        
+        # Add GOFLAGS after ARG TARGETOS
+        sed -i '/^ARG TARGETOS/a ARG GOFLAGS="-buildvcs=false"' deploy/Dockerfile
+        
+        # Remove any CMD line containing busybox
+        sed -i '/CMD.*busybox/d' deploy/Dockerfile
+
+        # Add safe final CMD
+        echo 'CMD ["/bin/sh", "-c", "echo noop"]' >> deploy/Dockerfile
+
+        git add deploy/Dockerfile
+        git diff --cached > ../goflags-fix.diff
+        git commit -m "Add GOFLAGS arg"
```

Comment:
What's the reason for these changes? They seem unnecessary to me.
The upstream Kaniko Dockerfile looks self-contained to me. Does it not build without those changes?